### DISCUSSION TITLE
revamped version of node-ts-cache

### DISCUSSION THREAD

**Discussion by boredland** [havsar/node-ts-cache#37] [Show and tell]
Hi there! As I didn't want to write something myself, I decided to give this a [bigger](https://github.com/ioki-mobility/node-ts-cache) overhaul.

The big changes so far:

- isLazy leads to returning the item and removing it afterwards
- items saved regularly are being saved with a redis ttl for the redis storage
- meta-data of fetched items is being returned (createdAt and if it has expired)
- coverage is being collected
- DI-style withCache factory to wrap your functions in
- afterExpired hook, so you can trigger recreations etc. if needed
- shouldCache hook, so you can decide if sth should be cached after a command ran
- dependencies cleansed
- testing using pure jest, no more assert&sinon
- superjson for json (de-)serialization
- removed lerna
- uses changeset for versioning
- tests for the elasticsearch module
- a very generic postgres module
- stuffed all tests into factories, so we can run them all for all storage modules (and fixes little issues I found doing that)
- removed isCachedForever
- moved to milliseconds for ttl

If you'd like this to be upstreamed - I'm very open to that!

**Comment by havsar** [comment]
@boredland I will have a look on this in the next days.
